File: DML/API_productos/src/main/java/ies/alcores/api_productos/model/Producto.java
package ies.alcores.api_productos.model;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToOne;
import lombok.Data;

import java.util.Date;

@Data
@Entity
public class Producto {

    @Id
    private String codBarras;
    private String marca;
    private String modelo;
    private int cantidad;
    private double precio;
    private Date fechaLanzamiento;
    @ManyToOne
    private Categoria categoria;

}

File: API_Producto/src/models/producto.ts
export class Producto {
    //atributos
    constructor(
        private _cantidad: number,
        private _codBarras: string,
        private _fechaLanzamiento: Date,
        private _marca: string,
        private _modelo: string,
        private _precio: number
    ){}

    get cantidad(): number {
        return this.cantidad;
    }

    set cantidad(value: number) {
        this.cantidad = value;
    }

    get marca(): string {
        return this._marca;
    }

    set marca(value: string) {
        this._marca = value;
    }       
}
